src/components/media-loader.js

```diff
@@ -46,17 +48,23 @@ AFRAME.registerComponent("media-loader", {
   async update() {
     try {
       const url = this.data.src;
+      const token = this.data.token;
 
-      this.showLoaderTimeout = setTimeout(() => {
-        const loadingObj = new THREE.Mesh(new THREE.BoxGeometry(), new THREE.MeshBasicMaterial());
-        this.el.setObject3D("mesh", loadingObj);
-        this.setShapeAndScale(true);
-      }, 100);
+      this.showLoaderTimeout =
+        this.showLoaderTimeout ||
+        setTimeout(() => {
```

Comment:
Maybe move this into `this.showLoader` so this wrapps a little more cleanly and avoids creating this funciton several times even though it densest need to close over anything.